### PR TITLE
Reworked to newer code and added scons for compiling.

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This now works it seems. I've tested the final changes on OSX and my previous tests should indicate it should still work on Windows as well (I'll try later). Someone will have to test on Linux (note, edit the SIMPLE.gdns file to add paths to the binary!!).

I've also changed the scons file so the libraries get placed in a bin subfolder instead of in src

### CODE REVIEW

**Comment by BastiaanOlij** [comment]
Note, after talking to @karroffel I've added support for the new API struct. The manual compile instructions and the scons file assume you are using https://github.com/GodotNativeTools/godot_headers which at the time of writing this has a slightly older structure causing some problems.

Until the repository is update point the include folder to your_godot_fork/modules/gdnative/include and include gdnative_api_struct.gen.h instead of gdnative_api_struct.h

File: c/SimpleDemo/src/simple.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include <godot.h>
+#include <gdnative_api_struct.h>
```
Comment (karroffel) [inline c/SimpleDemo/src/simple.c]: This should be `gdnative_api_struct.gen.h` 😛

Comment (BastiaanOlij) [reply c/SimpleDemo/src/simple.c]: Stop changing the interface :P 

I'll find some time to update this soon :)

**Comment by BastiaanOlij** [comment]
Ok, compiles again, had to make a few more changes to make VC happy.

**Pull request merged**